dev/com.ibm.ws.kernel.boot.common/src/io/openliberty/checkpoint/spi/CheckpointPhase.java

```diff
@@ -239,32 +240,19 @@ private synchronized boolean addHook(CheckpointHook hook, boolean multiThreaded,
         if (blockAddHooks) {
             return false;
         }
-        debug(() -> "Adding hook: " + hook + (multiThreaded ? " multi-threaded" : " single-threaded") + " rank: " + rank);
 
+        debug(() -> "Adding hook: " + hook + (multiThreaded ? " multi-threaded" : " single-threaded") + " rank: " + rank);
         Map<Integer, StaticCheckpointHook> addToHooks = multiThreaded ? multiThreadedHooks : singleThreadedHooks;
-        StaticCheckpointHook staticCheckpointHook = addToHooks.computeIfAbsent(rank, (o) -> {
-            final StaticCheckpointHook newHook = new StaticCheckpointHook(rank, multiThreaded);
-            if (context != null) {
-                Dictionary<String, Object> props = new Hashtable<>();
-                props.put(SERVICE_RANKING, Integer.valueOf(rank));
-                props.put(CheckpointHook.MULTI_THREADED_HOOK, Boolean.valueOf(multiThreaded));
-                registerHook(context, registerService, newHook, props);
-            }
-            return newHook;
-        });
+        StaticCheckpointHook staticCheckpointHook = addToHooks.computeIfAbsent(rank, (r) -> createStaticCheckpointHook(rank, multiThreaded));
```

Comment:
Do you think this solution should whine on requests to add a hook at a rank that is already mapped to a hook?  Today it will silently ignore the request. The developer must look for a missing call to `createStaticCheckpointHook()` in the trace.